data/microos/butane/script

```diff
@@ -4,25 +4,53 @@
 exec > >(exec tee -a /dev/console) 2>&1
 set -eux
 
-#
-### drives and filesystems
-#
-DRIVE='/dev/vdc'
-PART='/dev/vdc1'
+create_fs() {
+    DRIVE='/dev/vdc'
+    PART='/dev/vdc1'
+
+    # create partition and filesystem on additional drive
+    echo "label: gpt" | sfdisk "$DRIVE"
+    echo "name=testing_part" | sfdisk "$DRIVE" -N1
+
+    # label new partition for testing and create labeled ext4
+    mkfs.ext4 -L home "$PART"
+
+    # mount new partition
+    mount -t ext4 "$PART" /home
 
-# create partition and filesystem on additional drive
-echo "label: gpt" | sfdisk "$DRIVE"
-echo "name=testing_part" | sfdisk /dev/vdc -N1
+    # DEBUG
+    blkid
+    lsblk
+}
 
-# label new partition for testing and create labeled ext4
-mkfs.ext4 -L home "$PART"
+systemd_mount() {
+
+cat << EOF > /etc/systemd/system/home.mount
+[Unit]
+Before=local-fs.target
+Requires=systemd-fsck@dev-disk-by\x2dpartlabel-testing_part.service
+After=systemd-fsck@dev-disk-by\x2dpartlabel-testing_part.service
 
-# mount new partition
-mount -t ext4 "$PART" /home
+[Mount]
+Where=/home
+What=/dev/disk/by-partlabel/testing_part
+Type=ext4
 
-# DEBUG
-blkid
-lsblk
+[Install]
+RequiredBy=local-fs.target
+EOF
+
+}
+
+### set locale, keyboard and timezone
+# exception: sle-micro comes with symlink /etc/localtime, thus systemd-firstboot fails
```

Comment:
Does SLEM not have the `--force` option?

Reply:
It does but files to re-create /etc/localtime :/
http://kepler.suse.cz/tests/21920#step/disk_boot/7

Reply:
Ok, please add a comment. If this is reproducible in an updated system (the 5.4 GA medium might be too old), please open a bug report for that.

Reply:
https://bugzilla.suse.com/show_bug.cgi?id=1215618

BTW: After applying updates, `mkfs.ext4 -L home "$PART"` was failing reporting unknown fs.

Reply:
> BTW: After applying updates, mkfs.ext4 -L home "$PART" was failing reporting unknown fs.

I guess you installed updates with a combustion script? This won't work, kernel updates result in a broken system boot. (IIRC not documented yet. Should ideally be fixed, but it's unclear how)

Reply:
Nope, 
1) booted plain slem5.4
2) configured with wizard
3) registered + installed updates
4) removed `/boot/writable/firstboot_happened`
5) shutted down the VM and added combustion.qcow2
6) in grub added `combustion.firstboot`

Reply:
I suspect the initrd created by the update installation is `--hostonly` and the `ext4` module wasn't included. After `dracut -f --no-hostonly` it should work.

Reply:
Yup, that helped. Thanks!